src/Rings/msolve/interface.jl

```diff
@@ -0,0 +1,131 @@
+# we take a Singular ideal and extract the following data:
+# an int array lengths storing the lengths of each generator
+# an int array cfs storing the coefficients of each generator
+# an int array exps storing the exponent vectors of each generator
+function convert_singular_ideal_to_array(id::Singular.sideal)
+    ngens   = Singular.ngens(id)
+    nvars   = Singular.nvars(base_ring(id))
+    char    = Singular.characteristic(base_ring(id))
+    lens    = Int32[Singular.length(id[i]) for i in 1:ngens]
+    nterms  = sum(lens)
+    cfs     = Int32[]
+    exps    = Int32[]
+    if char ==  0
+        cfs = BigInt[]
+    else
+        cfs = Int32[]
+    end
+    for i = 1:Singular.ngens(id)
+        if char == 0
+            for c in Singular.coeffs(id[i])
+                push!(cfs, Singular.libSingular.n_GetMPZ(numerator(c).ptr, Singular.ZZ.ptr))
+                push!(cfs, Singular.libSingular.n_GetMPZ(denominator(c).ptr, Singular.ZZ.ptr))
+            end
+        else
+            for c in Singular.coeffs(id[i])
+                push!(cfs, Base.Int(c))
+            end
+        end
+        for e in Singular.exponent_vectors(id[i])
+            for j = 1:nvars
+                push!(exps, Base.Int(e[j]))
+            end
+        end
+    end
+    return lens, cfs, exps
+end
+
+# we know that the terms are already sorted and they are all different
+# coming from GroebnerBasis' F4 computation, so we do not need p_Add_q for
+# the terms, but we can directly set the next pointers of the polynomials
+function convert_ff_gb_array_to_singular_ideal(
+        bld::Int32,
+        blen::Array{Int32,1},
+        bcf::Array{Int32,1},
+        bexp::Array{Int32,1},
+        R::Singular.PolyRing
+        )
+    ngens = bld
+
+    nvars = Singular.nvars(R)
+    basis = Singular.Ideal(R, ) # empty ideal
+    # first entry in exponent vector is module component => nvars+1
```

Comment:
Interesting, I see you have `Singular.libSingular.p_SetExpV`, but in Singular.jl we have `p_SetExp` in a loop followed by `p_Setm`: https://github.com/oscar-system/Singular.jl/blob/master/src/poly/poly.jl#L1432 . What is the difference?